ui/src/main/java/com/wireguard/android/util/QrCodeFromFileScanner.kt

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2017-2022 WireGuard LLC. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.wireguard.android.util
+
+import android.content.ContentResolver
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.net.Uri
+import android.util.Log
+import com.google.zxing.BinaryBitmap
+import com.google.zxing.ChecksumException
+import com.google.zxing.DecodeHintType
+import com.google.zxing.FormatException
+import com.google.zxing.NotFoundException
+import com.google.zxing.RGBLuminanceSource
+import com.google.zxing.Reader
+import com.google.zxing.common.HybridBinarizer
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+/**
+ * Encapsulates the logic of scanning a barcode from a file,
+ * @property contentResolver - Resolver to read the incoming data
+ * @property reader - An instance of zxing's [Reader] class to parse the image
+ */
+class QrCodeFromFileScanner(
+    private val contentResolver: ContentResolver,
+    private val reader: Reader,
+) {
+
+    private fun scanBitmapForResult(source: Bitmap): Result<com.google.zxing.Result> {
```

Comment:
We don't use Kotlin's `Result` type in this codebase and I don't intend to start now, can we get rid of it? Throwing is perfectly fine, you can check in `TunnelImporter` how we handle catching and aggregation of errors.

Reply:
sure, do you still want me to throw and get rid of Result?

Reply:
Yeah I'd prefer we get rid of Result.

Reply:
Ok will do, I do realise that this matter of preference tho

Reply:
All done, switched to throw from Kotlin Result.

PS. The QR code lib returns 'Result' class, but it's not Kt Result class, same name, not same class